### PR TITLE
Nushi character preliminary implementation with ricochet movement

### PR DESCRIPTION
**Pull request by orhanbalci**
This is a preliminary implementation for nushi character described in #61. New behavior ricochet is added. Characters with ricochet behavior bounces off the world bounds. Initialization of singleton Nushi character is done in game initialization. Maybe this can be moved into a SingletonSpawner system.

### CODE REVIEW

**Comment by erlend-sh** [comment]
Thanks for the PR! We’re on the verge of shipping our v0.1.0 MVP release so it might be a few days until we can merge this in.

Was there anything about implementing a new creature that was particularly painful?

**Comment by orhanbalci** [comment]
Actually this was the first time I worked with ECS based framework. It enables developer to only focus on the real behavior your want.  The only think that I missed was to insert new component into CreaturePrefabData struct in order to be able to use it in prefab definition (.ron) files. This was due to my lack of knowledge. I needed to head back to documentation.

File: src/states/main_game.rs
Comment (khskarl) [inline src/states/main_game.rs]: Maybe we should have the wander system depend on the ricochet system? 
Gameplay wise this shouldn't make much of a difference, but I don't know how deterministic Specs' scheduler/dispatcher is, so maybe it's a good idea to make the order explicit.

Well, that's a small thing, but it still would be nice to have a second opinion :).

File: src/states/main_game.rs
Comment (khskarl) [inline src/states/main_game.rs]: Remember to remove these comments before merging :). White lines here and there would be nice for separating implicit the blocks of logic (such as the rng related logic, or transform related logic).

**Review by khskarl — APPROVED** [review]
It looks good to me! And thank you for the contribution 🎉. 

Sidenote
----- 
We finally have the most important creature, the bouncing DVD logo: https://bouncingdvdlogo.com/

File: src/states/main_game.rs
Comment (orhanbalci) [reply src/states/main_game.rs]: dependency moved to wander system.

File: src/states/main_game.rs
Comment (orhanbalci) [reply src/states/main_game.rs]: unnecessary commented lines removed

**Pull request merged**
